api/src/test/java/com/epam/pipeline/manager/pipeline/PipelineWithPermissionsTest.java

```diff
@@ -71,80 +67,264 @@
         expectedMap.put(pipeline1.getId(), permissionsForPipeline1);
         expectedMap.put(pipeline2.getId(), permissionsForPipeline2);
+
+
         assertPipelineWithPermissions(expectedMap);
 
-        grantPermissionsToPipeline(AclPermission.WRITE.getMask(), testUserSid2.getSid(), pipeline1.getId());
-        permissionsForPipeline1.put(TEST_OWNER2.toUpperCase(), 4);
+    }
+
+    @Test
+    public void shouldReturnPermissionsWhenGrantedToPipelineCase2() {
+        // key - pipelineId, value - <sid, mask>
+        final Map<Long, Map<String, Integer>> expectedMap = new HashMap<>();
+        final Map<String, Integer> permissionsForPipeline1 = new HashMap<>();
+        final Map<String, Integer> permissionsForPipeline2 = new HashMap<>();
+
+        final List<Pair<AbstractSecuredEntity, List<AbstractGrantPermission>>> pipelinesAndFolders = Arrays.asList(
+                new ImmutablePair<>(pipeline1, Collections.singletonList(new UserPermission(TEST_USER_1, AclPermission.WRITE.getMask()))),
+                new ImmutablePair<>(pipeline1, Collections.singletonList(new UserPermission(TEST_USER_2, AclPermission.WRITE.getMask()))),
+                new ImmutablePair<>(pipeline2, Collections.singletonList(new UserPermission(TEST_USER_1, AclPermission.READ.getMask()))),
+                new ImmutablePair<>(folder1, Collections.singletonList(new UserPermission(TEST_USER_1, AclPermission.READ.getMask()))),
+                new ImmutablePair<>(folder2, Collections.singletonList(new UserPermission(TEST_USER_1, AclPermission.READ.getMask()))));
+
+        initAclEntity(pipelinesAndFolders);
+
+        doReturn(Arrays.asList(pipeline1, pipeline2)).when(mockEntityManager).loadAllWithParents(eq(AclClass.PIPELINE), any(), any());
+        doReturn(Arrays.asList(folder1, folder2)).when(mockEntityManager).loadAllWithParents(eq(AclClass.FOLDER), any(), any());
+
+        final PipelineWithPermissions pipelineWithPermissions1 = new PipelineWithPermissions();
+        pipelineWithPermissions1.setId(pipeline1.getId());
+        pipelineWithPermissions1.setOwner(pipeline1.getOwner());
+        doReturn(pipelineWithPermissions1).when(mockPipelineWithPermissionsMapper).toPipelineWithPermissions(eq(pipeline1));
+
+        final PipelineWithPermissions pipelineWithPermissions2 = new PipelineWithPermissions();
+        pipelineWithPermissions2.setId(pipeline2.getId());
+        pipelineWithPermissions2.setOwner(pipeline2.getOwner());
+        doReturn(pipelineWithPermissions2).when(mockPipelineWithPermissionsMapper).toPipelineWithPermissions(eq(pipeline2));
+
+        final PipelineWithPermissions pipelineWithPermissions3 = new PipelineWithPermissions();
+        pipelineWithPermissions3.setId(pipeline2.getId());
+        pipelineWithPermissions3.setOwner(pipeline2.getOwner());
+        doReturn(pipelineWithPermissions3).when(mockPipelineWithPermissionsMapper).toPipelineWithPermissions(eq(pipeline2));
```

Comment:
Mocking of `pipelineWithPermissions2` is overriden by mocking of `pipelineWithPermissions3`. I would suggest to get rid of  `pipelineWithPermissions3` declaration at all because it doesn't really have any effect on the test.